src/terms/nonlocal.jl

```diff
@@ -178,22 +178,34 @@ end
 """
 Build form factors (Fourier transforms of projectors) for an atom centered at 0.
 """
-function build_form_factors(psp, qs)
-    qnorms = norm.(qs)
-    T = real(eltype(qnorms))
-    # Compute position-independent form factors
-    form_factors = zeros(Complex{T}, length(qs), count_n_proj(psp))
-    count = 1
-    for l in 0:psp.lmax, m in -l:l
-        prefac_lm = im^l .* ylm_real.(l, m, qs)
-        n_proj_l = size(psp.h[l + 1], 1)
+function build_form_factors(psp, G_plus_ks)
+    T = real(eltype(first(G_plus_ks)))
+
+    n_proj_max = maximum(l -> count_n_proj_radial(psp, l), 0:psp.lmax; init=0)
+    radials = IdDict{T,Matrix{T}}()
```

Comment:
clarify what this is in formula

Reply:
so maybe ps, so be consistent with psp_projector_fourier?

Reply:
That's how it was before, but it was a bit messy (Michael commented on that), and I think it's clear from the code that we're dealing with the projectors. I'll clarify in the comments.